packages/library-sync/src/androidMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncInitializer.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin.mongodb.internal
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.net.ConnectivityManager
+import android.net.ConnectivityManager.NetworkCallback
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkInfo
+import android.net.NetworkRequest
+import android.os.Build
+import androidx.startup.Initializer
+import io.realm.kotlin.internal.RealmInitializer
+import io.realm.kotlin.log.RealmLog
+
+/**
+ * An **initializer** for Sync specific functionality that does not fit into the `RealmInitializer`
+ * in cinterop.o allow Realm to access context properties.
+ */
+class RealmSyncInitializer : Initializer<Context> {
+
+    companion object {
+        @Suppress("DEPRECATION") // Should only be called below API 21
+        fun isConnected(cm: ConnectivityManager?): Boolean {
+            return cm?.let {
+                val networkInfo: NetworkInfo? = cm.activeNetworkInfo
+                networkInfo != null && networkInfo.isConnectedOrConnecting || isEmulator()
```

Comment:
Why are emulator always connected?

Reply:
No idea 😅 ... I pulled this code from somewhere else and has never really questioned it 🙈